nemo/collections/asr/parts/utils/offline_clustering.py

```diff
@@ -552,12 +550,13 @@ def eigDecompose(
             device = torch.cuda.current_device()
         laplacian = laplacian.float().to(device)
     else:
+        laplacian = laplacian.float().to(torch.device('cpu'))
         laplacian = laplacian.float()
     lambdas, diffusion_map = eigh(laplacian)
     return lambdas, diffusion_map
 
 
-def eigValueSh(laplacian: torch.Tensor, cuda: bool, device: torch.device = torch.device('cpu')) -> torch.Tensor:
+def eigValueSh(laplacian: torch.Tensor, cuda: bool, device: torch.device) -> torch.Tensor:
```

Comment:
why cuda and device? Isn't only one sufficient

Reply:
This was added long back because there are users setting cuda=True but device=cpu.
This is adding some flexibility to avoid errors on such cases.
If we need to remove this, lt requires a speparate PR since this involves whole diarization pipeline.